proposals/mqtt.md

```diff
@@ -0,0 +1,83 @@
+# [MQTT](https://mqtt.org/) Discovery Handler
+
+## MQTT and its use
+
+MQTT is a simple message bus with a broad use in IoT devices, such device will usually connect to an existing MQTT broker on the network.
+This allows reuse of the same broker for multiple devices.
+
+Let's quickly introduce some key concepts of MQTT:
+
+- An MQTT message consists of a slash (`/`) separated UTF-8 topic and a completely free-form payload with no built-in notion of sender
+- Topics are only used for message routing and are not "created" or "destroyed"
+- By default, messages are not stored, and a message with no client subscribing to its topic is simply discarded
+- There are no specified way of querying connected clients or current subscriptions
+- An MQTT client can use wildcards when subscribing to topics
+
+Because of these properties, it is impossible to discover devices that do not publish any message.
+
+Usually a device will publish into topics that are specific to that device instance so that it becomes possible to know the origin of the message.
+For example a weather station might publish into topics like `weather/<city>/temperature` and `weather/<city>/pressure` with `<city>` the location of station.
```

Comment:
Wanted to point out a coAP discovery handler that was worked on awhile ago as that protocol has topics and might provide some more inspiration. Proposal is `coap.md` in the closed pr: https://github.com/project-akri/akri/pull/346/files#diff-0c3abadcc72d01ab2b22fd1f4a2e696f0f3c350909e2d991cf7fea22a4311789